### DISCUSSION TITLE
Allow TreeNodes to have CSS classes

### DISCUSSION THREAD

**Discussion by KalleDK** [Textualize/textual#5293] [Ideas]
I'm writing a program that uses the tree to navigate, and I would love to be able to have classes on the nodes in such a way that I can mark all "warning" nodes with a color or all "ok" nodes woth another. Right now I prepend an emoji, but it gets cluttery very fast.

I have a workaround right now with Text.assemble("label", style="red")... But this does not work well with the css afaik (also it would be lovely to use the theme colors instead of hardcoding)

**Comment by KalleDK** [comment]
Or as minimal viable allow me to apply a color like $warning for  the label

**Comment by darrenburns** [comment]
You can access the theme variables in the `App.theme_variables` attribute. This is a dict which maps the variable name to the value it holds.

So from a widget, `self.app.theme_variables.get("text-warning")` will retrieve the `$text-warning` variable (which is legible when applied as a text colour across all themes, so you probably want that instead of `$warning`).

**Reply by KalleDK** [reply]
And that can be applied via the Text.assemble? And it updates when the user changes theme?

**Reply by darrenburns** [reply]
Yes, it just returns a string, so it'll work anywhere strings like `"red"` and `"#ff0000"` work, including in `Text.assemble`.

The value will update when the user changes theme, yeah.

**Reply by KalleDK** [reply]
I see - but the nodes needs to be "updated" when the user changes theme via the control palette. But it is still way better then what I have :)
